utils/src/main/kotlin/ArchiveUtils.kt

```diff
@@ -37,39 +38,62 @@ import org.apache.commons.compress.compressors.bzip2.BZip2CompressorInputStream
 import org.apache.commons.compress.compressors.gzip.GzipCompressorInputStream
 
 private val UNCOMPRESSED_EXTENSIONS = listOf(".pom")
-private val TAR_EXTENSIONS = listOf(".gem", ".tar", ".tar.gz", ".tgz", ".tar.bz2", ".tbz2")
+private val TAR_EXTENSIONS = listOf(".gem", ".tar")
+private val TAR_BZIP2_EXTENSIONS = listOf(".tar.bz2", ".tbz2")
+private val TAR_GZIP_EXTENSIONS = listOf(".tar.gz", ".tgz")
 private val ZIP_EXTENSIONS = listOf(".aar", ".egg", ".jar", ".war", ".whl", ".zip")
 private val SEVENZIP_EXTENSIONS = listOf(".7z")
 
-val ARCHIVE_EXTENSIONS = TAR_EXTENSIONS + ZIP_EXTENSIONS + SEVENZIP_EXTENSIONS
+val ARCHIVE_EXTENSIONS = TAR_EXTENSIONS + TAR_BZIP2_EXTENSIONS + TAR_GZIP_EXTENSIONS + ZIP_EXTENSIONS +
+        SEVENZIP_EXTENSIONS
 
+/**
+ * Unpack the [File] to [targetDirectory].
+ */
 fun File.unpack(targetDirectory: File) {
     val lowerName = name.toLowerCase()
+    if (SEVENZIP_EXTENSIONS.any { lowerName.endsWith(it) }) {
+        unpack7Zip(targetDirectory)
+    } else {
+        inputStream().unpack(name, targetDirectory)
+    }
+}
+
+/**
+ * Unpack the [InputStream] to [targetDirectory]. The compression scheme is guessed from the [fileName].
+ */
+fun InputStream.unpack(fileName: String, targetDirectory: File) {
+    val lowerName = fileName.toLowerCase()
     when {
         UNCOMPRESSED_EXTENSIONS.any { lowerName.endsWith(it) } -> {
+            // Just copy the stream to the target file.
+            use { File(targetDirectory, fileName).writeBytes(it.readBytes()) }
```

Comment:
Please use `InputStream.copyTo` instead of `r/w bytes`.

Reply:
Done.